editor/src/core/property-controls/property-controls-local-parser-bridge.ts

```diff
@@ -0,0 +1,100 @@
+import { Either, isRight, left, mapEither, right } from '../shared/either'
+import {
+  ArbitraryJSBlock,
+  clearJSXElementUniqueIDs,
+  JSXElementChild,
+  JSXElementWithoutUID,
+  UtopiaJSXComponent,
+} from '../shared/element-template'
+import { jsxSimpleAttributesToProps } from '../shared/jsx-attributes'
+import { objectMapDropNulls } from '../shared/object-utils'
+import { Imports, isParseFailure, isParseSuccess } from '../shared/project-file-types'
+import { createParseFile, getParseResult, UtopiaTsWorkers } from '../workers/common/worker-types'
+
+type ProcessedParseResult = Either<
+  string,
+  { importsToAdd: Imports; elementToInsert: JSXElementWithoutUID }
+>
+const resultCache: Map<string, ProcessedParseResult> = new Map()
+
+export async function getCachedParseResultForUserStrings(
```

Comment:
Please can you move this file out of `property-controls`, since this code is purely related to communication with the parser-printer and so should live closer to that